modules/tensor_mechanics/src/kernels/StressDivergenceTruss.C

```diff
@@ -0,0 +1,154 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "StressDivergenceTruss.h"
+
+// MOOSE includes
+#include "Assembly.h"
+#include "Material.h"
+#include "MooseVariable.h"
+#include "SystemBase.h"
+#include "RankTwoTensor.h"
+#include "NonlinearSystem.h"
+#include "MooseMesh.h"
+
+#include "libmesh/quadrature.h"
+
+registerMooseObject("TensorMechanicsApp", StressDivergenceTruss);
+
+InputParameters
+StressDivergenceTruss::validParams()
+{
+  InputParameters params = Kernel::validParams();
+  params.addClassDescription("Quasi-static and dynamic stress divergence kernel for truss element");
+  params.addRequiredParam<unsigned int>(
+      "component",
+      "An integer corresponding to the direction the variable this kernel acts in. (0 for disp_x, "
+      "1 for disp_y, 2 for disp_z)");
+  params.addRequiredCoupledVar(
+      "displacements",
+      "The displacements appropriate for the simulation geometry and coordinate system");
+  params.set<bool>("use_displaced_mesh") = true;
+  return params;
+}
+
+StressDivergenceTruss::StressDivergenceTruss(const InputParameters & parameters)
+  : Kernel(parameters),
+    _component(getParam<unsigned int>("component")),
```

Comment:
We should transition to determining the component automatically from the position of `_var.name()`  in `displacements`

Reply:
Since none of the other mechanics kernels do this, I would suggest we leave this as-is for now, and then go back and change this for all of them at the same time.